File: lib/src/main/java/stringtable/Path.java
package stringtable;

public class Path {

	public static String combine(String pathString, String childName) {
		String pathDelimiter = getDelimiter(pathString + childName);

		String result = pathString;

		if (pathString.endsWith(pathDelimiter))
			result = pathString;
		else
			result = pathString + pathDelimiter;

		if (childName.startsWith(pathDelimiter))
			result = result + childName.substring(1);
		else
			result = result + childName;

		return result;
	}

	private static String getDelimiter(String path) {
		if (path.indexOf('\\') != -1)
			return "\\";
		else
			return "/";
	}

	public static String combine(String pathString, String childName1, String... filenames) {
		String result = combine(pathString, childName1);

		for (String filename : filenames) {
			result = combine(result, filename);
		}
		return result;
	}
}

File: src/main/java/com/rsupport/stringtable/Path.kt
package com.rsupport.stringtable

object Path {
    private fun combine(pathString: String, childName: String): String {
        val pathDelimiter = getDelimiter(pathString + childName)
        var result: String = if (pathString.endsWith(pathDelimiter)) pathString else pathString + pathDelimiter
        result = if (childName.startsWith(pathDelimiter)) result + childName.substring(1) else result + childName
        return result
    }

    private fun getDelimiter(path: String): String {
        return if (path.indexOf('\\') != -1) "\\" else "/"
    }

    fun combine(pathString: String, childName1: String, vararg filenames: String): String {
        var result = combine(pathString, childName1)
        for (filename in filenames) {
            result = combine(result, filename)
        }
        return result
    }
}